File: app/src/main/java/lishui/study/ui/SquareFragment.java
package lishui.study.ui;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProviders;

import com.bumptech.glide.Glide;

import lishui.study.R;
import lishui.study.common.BaseFragment;
import lishui.study.databinding.SquareFragmentBinding;
import lishui.study.viewmodel.SquareViewModel;

public class SquareFragment extends BaseFragment {

    private SquareFragmentBinding mBinding;

    private SquareViewModel mViewModel;

    public static SquareFragment newInstance() {
        return new SquareFragment();
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        mBinding = DataBindingUtil.inflate(inflater, R.layout.square_fragment, container, false);
        return mBinding.getRoot();
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        mViewModel = ViewModelProviders.of(this).get(SquareViewModel.class);
        subscribeToViewModel(mViewModel);

        mViewModel.loadThumbnailBitmap();
    }

    private void subscribeToViewModel(SquareViewModel viewModel) {
        viewModel.getThumbnailLiveData().observe(getViewLifecycleOwner(), bitmap -> {
            if (bitmap != null) {
                Glide.with(this).load(bitmap).into(mBinding.firstThumbnail);
//                mImageView.setImageBitmap(bitmap);
            }
        });
    }

}

File: app/src/main/java/lishui/study/ui/SquareFragment.kt
package lishui.study.ui

import android.content.ComponentName
import android.content.Context
import android.content.Intent
import android.content.pm.ShortcutInfo
import android.content.pm.ShortcutManager
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.viewModels
import lishui.study.R
import lishui.study.common.BaseFragment
import lishui.study.databinding.SquareFragmentBinding
import lishui.study.viewmodel.SquareViewModel

class SquareFragment : BaseFragment() {

    private lateinit var mBinding: SquareFragmentBinding
    private val mViewModel by viewModels<SquareViewModel>()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        mBinding = DataBindingUtil.inflate(inflater, R.layout.square_fragment, container, false)
        return mBinding.root
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        mBinding.bitmapLiveData = mViewModel.thumbnailLiveData
        mBinding.lifecycleOwner = this
        mViewModel.loadThumbnailBitmap()

        mBinding.firstThumbnail.setOnClickListener {
            val shortcutManager = context?.getSystemService(Context.SHORTCUT_SERVICE) as ShortcutManager
            if (shortcutManager.isRequestPinShortcutSupported) {
                val pinShortcutInfo = ShortcutInfo.Builder(context, "test-shortcut").apply {
                    val intent = Intent(context, SearchActivity::class.java)
                    intent.action = "lishui.study.action.SEARCH"
                    setIntent(intent)
                    setActivity(ComponentName("lishui.study", "lishui.study.ui.MainActivity"))
                    setShortLabel("test short label")
                }.build()

                Log.d("lishuii", "pinShortcutInfo=$pinShortcutInfo")
                shortcutManager.requestPinShortcut(pinShortcutInfo, null)
            }
        }
    }
}